dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java

```diff
@@ -752,7 +753,7 @@ public void setDateProperty(com.dotcms.contenttype.model.field.Field field, Date
 	 */
 	public Date getDateProperty(String fieldVarName) throws DotRuntimeException {
 		try{
-			return (Date)map.get(fieldVarName);
+			return map.containsKey(fieldVarName) ? (Date) map.get(fieldVarName) : null;
 		}catch (Exception e) {
 			 throw new DotRuntimeException("Unable to retrive field value", e);
```

Comment:
Not related to your specific changes, but can you add the name of the Date field to the error message please?

Reply:
done https://github.com/dotCMS/core/pull/22096/commits/fc2bbe074dc78a05ae755a49843d0b9d4662702a#diff-1367236d40a612b923ececfa521053143f026f60f035c9f9963aae4efdfaf034R758